compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/classpathDiff/ClasspathSnapshotShrinker.kt

```diff
@@ -174,27 +174,36 @@ private fun ClasspathSnapshot.getNonDuplicateClassSnapshots(): List<ClassSnapsho
 
 /** Used by [shrinkAndSaveClasspathSnapshot]. */
 private sealed class ShrinkMode {
-    object NoChanges : ShrinkMode()
+    object UnchangedLookupsUnchangedClasspath : ShrinkMode()
 
-    class IncrementalNoNewLookups(
-        val shrunkCurrentClasspathAgainstPreviousLookups: List<AccessibleClassSnapshot>,
+    class UnchangedLookupsChangedClasspath(
+        val currentClasspathSnapshot: List<AccessibleClassSnapshot>,
+        val shrunkCurrentClasspathAgainstPreviousLookups: List<AccessibleClassSnapshot>
     ) : ShrinkMode()
 
-    class Incremental(
+    sealed class ChangedLookups : ShrinkMode() {
+        abstract val addedLookupSymbols: Set<LookupSymbolKey>
+    }
+
+    class ChangedLookupsUnchangedClasspath(
+        override val addedLookupSymbols: Set<LookupSymbolKey>
+    ) : ChangedLookups()
+
+    class ChangedLookupsChangedClasspath(
+        override val addedLookupSymbols: Set<LookupSymbolKey>,
         val currentClasspathSnapshot: List<AccessibleClassSnapshot>,
-        val shrunkCurrentClasspathAgainstPreviousLookups: List<AccessibleClassSnapshot>,
-        val addedLookupSymbols: Set<LookupSymbolKey>
-    ) : ShrinkMode()
+        val shrunkCurrentClasspathAgainstPreviousLookups: List<AccessibleClassSnapshot>
+    ) : ChangedLookups()
 
     object NonIncremental : ShrinkMode()
 }
 
 internal fun shrinkAndSaveClasspathSnapshot(
     classpathChanges: ClasspathChanges.ClasspathSnapshotEnabled,
     lookupStorage: LookupStorage,
-    currentClasspathSnapshot: List<AccessibleClassSnapshot>?, // Not null iff classpathChanges is ToBeComputedByIncrementalCompiler
-    shrunkCurrentClasspathAgainstPreviousLookups: List<AccessibleClassSnapshot>?, // Same as above
-    metrics: BuildMetricsReporter
+    precomputedCurrentClasspathSnapshot: List<AccessibleClassSnapshot>?, // Not null iff classpathChanges is ToBeComputedByIncrementalCompiler
```

Comment:
nit: Consider keeping the old name as it is informative and shorter. The same below.

Reply:
Yeah there was a name clash with a local variable below, but I've now kept the old name here and renamed that local variable instead.